addons/account/models/account_account.py

```diff
@@ -902,7 +921,19 @@ def create(self, vals_list):
                     start_code = prefix.ljust(digits - 1, '0') + '1' if len(prefix) < digits else prefix
                     vals['code'] = self.with_company(company)._search_new_account_code(start_code, cache)
                     cache.add(vals['code'])
-            records_list.append(super(AccountAccount, self.with_company(company)).create(vals_list_for_company))
+                elif 'code' not in vals and 'code_by_company' in vals:
+                    vals['code'] = vals['code_by_company'][company.id]
+            code_by_company_list = [vals.pop('code_by_company', {}) for vals in vals_list_for_company]
+            # Create the accounts with a single company and a single code.
+            new_accounts = super(AccountAccount, self.with_company(company)).create(vals_list_for_company)
+            # Add the other codes and companies on each account.
+            for new_account, code_by_company in zip(new_accounts, code_by_company_list):
+                for company_id, code in code_by_company.items():
+                    if company_id != company.id:
+                        new_account.with_context({'allowed_company_ids': [company_id, company.id]}).code = code
+                if other_companies:
+                    new_account.write({'company_ids': [Command.link(c.id) for c in other_companies]})
```

Comment:
I tend to do it just for variables in generator expressions or arguments of lambda functions, I personally think it's nice because it lets you know that the variable doesn't come from the surrounding scope. I've changed it as you suggested here (as I think the full variable name is very readable). Thanks!